libs/core/src/lib/token/tokenizer.component.ts

```diff
@@ -0,0 +1,93 @@
+    ViewEncapsulation
+} from '@angular/core';
+import { FormControlDirective } from '../form/form-control/form-control.directive';
+import { TokenComponent } from './token.component';
+
+@Component({
+  selector: 'fd-tokenizer',
+  templateUrl: './tokenizer.component.html',
+  styleUrls: ['./tokenizer.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TokenizerComponent implements AfterViewInit {
+
+    /** @hidden */
+    @ContentChildren(forwardRef(() => TokenComponent))
+    tokenList: QueryList<TokenComponent>;
+
+    /** @hidden */
+    @ContentChild(forwardRef(() => FormControlDirective), {static: true})
+    input: FormControlDirective;
+
+    /** Used to add focus class to the tokenizer-example */
+    @Input()
+    tokenizerHasFocus: boolean = false;
+
+    /** Whether the tokenizer-example is compact */
+    @Input()
+    compact: boolean = false;
+
+    /** The value for the tokenizer input */
+    @Input()
+    inputValue: string;
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        if (this.input && this.input.elementRef) {
+            this.input.elementRef.nativeElement.addEventListener('keydown', (event) => {
+                this.handleKeyDown(event, this.tokenList.length);
+            });
+        }
+    }
+
+    /** @hidden */
+    handleKeyDown(event: KeyboardEvent, fromIndex: number): void {
+        let newIndex;
+        if (event.code === 'ArrowLeft') {
+            newIndex = fromIndex - 1;
+        } else if (event.code === 'ArrowRight') {
+            newIndex = fromIndex + 1;
+        }
+        if (newIndex === this.tokenList.length && event.code === 'ArrowRight') {
+            this.input.elementRef.nativeElement.focus();
+        } else if (newIndex || newIndex === 0) {
+            this.focusTokenElement(event, newIndex);
+        }
+    }
+
+    /** @hidden */
+    focusTokenElement(event: KeyboardEvent, newIndex: number): HTMLElement {
+        // function needs to be defined in order to be referenced later by addEventListener/removeEventListener
+        const handleFunctionReference = (e) => {
+            if (newIndex || newIndex === 0) {
+                this.handleKeyDown(e, newIndex);
+            }
+        };
+        let elementToFocus;
+        if (newIndex >= 0 && newIndex < this.tokenList.length) {
+            elementToFocus = this.tokenList.filter((element, index) => index === newIndex)[0]
+                .elementRef.nativeElement.querySelector('.fd-token');
+            // element needs tabindex in order to be focused
+            elementToFocus.setAttribute('tabindex', '0');
+            elementToFocus.focus();
+            elementToFocus.addEventListener('keydown', handleFunctionReference);
+            elementToFocus.addEventListener('blur', () => {
+                elementToFocus.setAttribute('tabindex', '-1');
+                elementToFocus.removeEventListener('keydown', handleFunctionReference);
```

Comment:
Could you put it to different method, named for example: `addKeyboardListener(element: type): void`